clients/rust/README.md

```diff
@@ -0,0 +1,276 @@
+    amount,
+    authorization_data: None,
+};
+
+// instruction accounts
+let cpi_transfer = TransferV1Cpi {
+    __program: metadata_program_info,
+    __args: args,
+    token: owner_token_info,
+    token_owner: owner_info,
+    destination_token: destination_token_info,
+    destination_owner: destination_info,
+    mint: mint_info,
+    metadata: metadata_info,
+    authority: vault_info,
+    payer: payer_info,
+    system_program: system_program_info,
+    sysvar_instructions: sysvar_instructions_info,
+    spl_token_program: spl_token_program_info,
+    spl_ata_program: spl_ata_program_info,
+    edition: edition_info,
+    token_record: None,
+    destination_token_record: None,
+    authorization_rules: None,
+    authorization_rules_program: None,
+};
+
+// performs the CPI
+cpi_transfer.invoke_signed(&[&signer_seeds])
+```
+
+You can also use the `TransferV1CpiBuilder` to simplify the process:
+```rust
+let cpi_transfer = TransferV1CpiBuilder::new(metadata_program_info)
+    .token(owner_token_info)
+    .token_owner(owner_info)
+    .destination_token(destination_token_info)
+    .destination_owner(destination_info)
+    .mint(mint_info)
+    .metadata(metadata_info)
+    .edition(edition_info)
+    .authority(vault_info)
+    .payer(payer_info)
+    .system_program(system_program_info)
+    .sysvar_instructions(sysvar_instructions_info)
+    .spl_token_program(spl_token_program_info)
+    .spl_ata_program(spl_ata_program_info)
+    .amount(amount)
+    .build();
+
+// performs the CPI
+cpi_transfer.invoke_signed(&[&signer_seeds])
+```
+
+> **Note**
+> `*Builder` provide a simplified way to create the required structs, since they take advantage of any default value set on the Kinobi config and do not require to set a `None` value to optional fields.
+
+## PDA helpers
+
+Account types (e.g., `Metadata`) have associated functions to find PDA or to create PDA `Pubkey`s:
+```rust
+impl Metadata {
+    pub fn find_pda(mint: Pubkey) -> (solana_program::pubkey::Pubkey, u8) {
+        solana_program::pubkey::Pubkey::find_program_address(
+            &[
+                "metadata".as_bytes(),
+                crate::MPL_TOKEN_METADATA_ID.as_ref(),
+                mint.as_ref(),
+            ],
+            &crate::MPL_TOKEN_METADATA_ID,
+        )
+    }
+    pub fn create_pda(
+        mint: Pubkey,
+        bump: u8,
+    ) -> Result<solana_program::pubkey::Pubkey, solana_program::pubkey::PubkeyError> {
+        solana_program::pubkey::Pubkey::create_program_address(
+            &[
+                "metadata".as_bytes(),
+                crate::MPL_TOKEN_METADATA_ID.as_ref(),
+                mint.as_ref(),
+                &[bump],
+            ],
+            &crate::MPL_TOKEN_METADATA_ID,
+        )
+    }
+}
+```
+> If a bump seed is known, it is _cheaper_ (in terms of compute units) to use the `create_pda` function, in particular for on-chain code.
+
+ ## Why using a client library (SDK)?
```

Comment:
imho this section should go at the top before the examples, just because it really sells the client right off the bat and its not too long of a section so it can easily be seen that the examples would be right after it.